File: app/src/main/java/com/anthonymandra/image/JpegConfiguration.java
package com.anthonymandra.image;

public class JpegConfiguration extends ImageConfiguration {
    private int quality;

    @Override
    public ImageType getType() {
        return ImageType.jpeg;
    }

    @Override
    public String getExtension() {
        return "jpg";
    }

    @Override
    protected void parse(String config) {
        quality = Integer.parseInt(config);
    }

    @Override
    protected String convertToPreference() {
        return String.valueOf(quality);
    }

    public int getQuality() {
        return quality;
    }

    public void setQuality(int quality) {
        this.quality = quality;
    }
}

File: app/src/main/java/com/anthonymandra/image/JpegConfiguration.kt
package com.anthonymandra.image

class JpegConfiguration(val quality: Int = 0) : ImageConfiguration() {	// TODO: 0 is not a good default
	override val parameters = quality.toString()
	override val type = ImageConfiguration.ImageType.jpeg
	override val extension ="jpg"

	constructor(preference: String) : this(preference.toInt())
}
